core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -254,9 +255,9 @@ class JDBCSourceAsColumnarStore(private var _connProperties: ConnectionPropertie
             // or it will be committed later
             val txId = SparkShellRDDHelper.snapshotTxIdForWrite.get
             if (txId != null && !txId.equals("null")) {
-              val statement = conn.createStatement()
-              statement.execute(
-                s"call sys.GET_SNAPSHOT_TXID('$txId')")
+              val statement = conn.prepareCall("call sys.GET_SNAPSHOT_TXID(?)")
+              statement.registerOutParameter(1, java.sql.Types.VARCHAR)
+              statement.execute()
```

Comment:
Yes it fails in execution with SQLState.LANG_DB2_PARAMETER_NEEDS_MARKER (42886) saying that parameter should be a marker '?' and not a value. The problem is that regardless of whether output value is used or not, the implementation of GfxdSystemProcedures.GET_SNAPSHOT_TXID needs an array which it will fill out that will only happen if registerOutParameter has been called.

If you would have registered GET_SNAPSHOT_TXID as a zero-arg function returning a string rather than a procedure with output parameter, then could have invoked it with "VALUES GET_SNAPSHOT_TXID()" which would have been cleaner.